core/domain/prod_validators.py

```diff
@@ -3103,6 +3103,29 @@ def _get_external_id_relationships(cls, item):
         return field_name_to_external_model_references
 
 
+class ReviewerAndSuggestionCountsModelValidator(
+        base_model_validators.BaseModelValidator):
+    """Class for validating ReviewerAndSuggestionCountsModel."""
+
+    @classmethod
+    def _get_model_id_regex(cls, unused_item):
+        # Since this is a singleton model, it has only one valid ID:
+        # reviewer_and_suggestion_counts.
+        return '^%s$' % (
+            suggestion_models.REVIEWER_AND_SUGGESTION_COUNTS_MODEL_ID)
+
+    @classmethod
+    def _get_external_id_relationships(cls, item):
+        return []
+
+    @classmethod
+    def _get_model_domain_object_instance(cls, item):
+        return (
+            suggestion_services
+            .create_reviewer_and_suggestion_counts_from_model(item)
+        )
+
```

Comment:
Do we need to have model integeration test? like the count of reviewers counts matches the  `UserContributionRightsModel`, open suggestion count matches the count in the `ReviewerAndSuggestionCountsModel` etc.

Reply:
How would that work? Just make helpers in the storage class to get the counts and check that way? Then there would be at least O(2N) storage calls, where N is the number of language codes that translation suggestions are offered in. Alternatively, I'm currently making a MR job to populate the fields, could that MR job be called after this one to ensure the counts are correct instead of adding the model integration test?

Reply:
It's fine if it takes O(2N) clalls (in the optimal structure) as these checks are going to run in the background and we only have one stats model. We need to write this models integration validation checks in prod_validators file, you can check [the wiki page](https://github.com/oppia/oppia/wiki/Writing-Validators-for-storage-models) written by @ankita240796, if required you can ask her for help! :)

Reply:
Thanks @DubeySandeep! I wrote the checks to make sure the counts are correct in the follow up PR :)